src/Mod/Start/App/DisplayedFilesModel.cpp

```diff
@@ -190,16 +190,50 @@ void DisplayedFilesModel::addFile(const QString& filePath)
     if (!qfi.isReadable()) {
         return;
     }
+
+    // Check if FreeCAD can open the file by its extension

```

Comment:
The code below is self-explanatory enough to make this comment obsolete. The same applies to the comments up to line 204